src/zenml/orchestrators/utils.py

```diff
@@ -50,6 +51,35 @@ def create_tfx_pipeline(
     )
 
 
+def is_cached_step(execution_info: data_types.ExecutionInfo) -> bool:
+    """Returns the caching status of a tfx step.
+
+    Args:
+        execution_info: The execution info of a tfx step.
+
+    Returns:
+        The caching status of a tfx step as a boolean value.
+    """
+    repository = Repository()
+    metadata_store = repository.get_stack(
+        repository.active_stack_name
+    ).metadata_store
+
+    step_name = str(
+        execution_info.exec_properties["zenml-pipeline_parameter_name"]
+    )[1:-1]
+    pipeline_name = execution_info.pipeline_info.id  # type: ignore [attr-defined]
+    pipeline_run_name = execution_info.pipeline_run_id
+
+    status = (
+        metadata_store.get_pipeline(pipeline_name)  # type: ignore [union-attr]
+        .get_run(pipeline_run_name)  # type: ignore [arg-type]
```

Comment:
Any way we can avoid these `type: ignore` here? @schustmi perhaps any tips?

Reply:
Yeah actually we could define a mypy ignore file where we could define separately which lines or statements could be ignored for certain classes of errors, but I think that's a) a slippy slope and b) too many things change too often in our codebase for that to work.

The problem (I think) is how `tfx` type annotate their outputs, which then we have to handle. One way of getting around it would be (at the top of the file) to turn off the strict errors for whenever we're not handling a potential `None` value that comes out of `tfx` in that chain. Not sure whether that's a good idea or not.

Reply:
I think the ignores here come from the fact that
- `metadata_store.get_pipeline(...)` returns an optional pipeline, so we would have to extract that and then check if the pipeline actually exists
- `pipeline_run_name` probably isn't a string, so we'd have to check beforehand to see that it's actually a string

So it would make the code a few lines longer due to these extra checks but at the same time safer and we can raise more explicit errors

Reply:
Done